### PR TITLE
Data Migration: Create some testing infrastructure for testing data migration solutions

### PR DESCRIPTION
**Pull request by tyler-cai-microsoft**
[AB#20657](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/20657)

Build some basic testing infrastructure for data migration to save us some time from having to write the tests over and over again. Obviously, these tests can be iterated over and are not complete, but they should be a good baseline starting point.

### CODE REVIEW

File: packages/test/test-end-to-end-tests/src/test/data-migration/basicMigration.spec.ts [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+	name: string;
+	runtimeFactory: ContainerRuntimeFactoryWithDefaultDataStore;
+	migrateWithSummary(provider: ITestObjectProvider): Promise<string>;
+	migrateWithoutSummary(provider: ITestObjectProvider): Promise<RootDO2>;
+	migrateWithManyContainers(...containers: IContainer[]): Promise<RootDO2[]>;
+}
+
+const migrationStrategies: IMigrationStrategy[] = [];
+
+describeCompat.skip("basicMigration", "NoCompat", (getTestObjectProvider) => {
```
Comment (anthony-murphy) [inline packages/test/test-end-to-end-tests/src/test/data-migration/basicMigration.spec.ts:118]: would it make more sense to put these in the local-server or functional test packages, which are easier to use the real types in, as they do not use compat. additionally, they don't need to run against multiple servers, which tends to make the tests ever more complicated. 

Eventually we probably want to run against real services, but might be premature now.

Comment (tyler-cai-microsoft) [reply packages/test/test-end-to-end-tests/src/test/data-migration/basicMigration.spec.ts:118]: Sounds good to me

Comment (tyler-cai-microsoft) [reply packages/test/test-end-to-end-tests/src/test/data-migration/basicMigration.spec.ts]: Updated

Comment (tyler-cai-microsoft) [reply packages/test/test-end-to-end-tests/src/test/data-migration/basicMigration.spec.ts]: As a side note, I think I would also want to run these against tinylicious as well.

**Review by jatgarg — APPROVED** [review]

**Pull request merged**